### PR TITLE
Dark mode

### LINKED ISSUE #555: docs: Add a dark mode
I just realized in the mad dash to get the new website out the door in March, we didn't get around to implementing a dark mode, and then I forgot to file an issue for that. So here it is! 😂

### PR DESCRIPTION
**Pull request by vvveebs**
This is a 🙋 feature or enhancement.

## Summary

Adds a dark mode to the Bridgetown website.

## Context

Closes #555

This implementation mainly reuses existing colors. An exception is made for meeting color contrast accessibility requirements though - see inline PR comment.

The styling will of course need various changes to match an actual design but this should hopefully form a basis.

Code syntax styling may be an area that needs extra attention.

## Screenshots

|  Light Mode  | Dark Mode |
| ------------- | ------------- |
| image | image|

### CODE REVIEW

**Comment by jaredcwhite** [comment]
@vvveebs This is fantastic! And going the extra mile to add the theme picker too. 👍🏻👍🏻

I'll review in more detail soon…would be awesome to roll this out along with Bridgetown 1.2.

File: bridgetown-website/frontend/styles/theme-dark.css [hunk 1]
```diff
@@ -0,0 +1,93 @@
+:root.theme-dark {
+  --body-background: var(--color-darkest-gray);
+  --body-color: var(--color-light-gray);
+
+  --sl-color-primary-700: var(--color-light-green);
+  --sl-color-primary-600: var(--color-lighter-green);
+  --sl-color-primary-500: var(--color-lightest-green);
```
Comment (vvveebs) [inline bridgetown-website/frontend/styles/theme-dark.css:7]: A couple more lighter green colors were added here for functional reasons. 

Testing the site with Lighthouse in Chrome's devtools showed issues with color contrast for primary buttons against a dark background.

I expect these will need some design thought - they look out of sorts but I wasn't sure how else to approach this.

Example of these colors for a button and its hover state:

image

image

Comment (jaredcwhite) [reply bridgetown-website/frontend/styles/theme-dark.css:7]: Hmm, yeah we'll need to think about the primary button style…either retain a dark green background with white text (perhaps with thin <=1px border around it), or create a more saturated green with black text. That particular example seems lackluster

Comment (vvveebs) [reply bridgetown-website/frontend/styles/theme-dark.css:7]: 👍  Had a play around with these ideas..  let me know if you want me to push one of these as a commit.

image

image

Comment (jaredcwhite) [reply bridgetown-website/frontend/styles/theme-dark.css:7]: I like the dark with the outline (the first option)! In general it seems like dark modes which also offer better accessibility tend to increase outline contrast. 👍

Comment (vvveebs) [reply bridgetown-website/frontend/styles/theme-dark.css]: PR updated!

File: bridgetown-website/src/_components/shared/navbar.serb [hunk 2]
```diff
@@ -51,6 +51,7 @@
 
         <kbd>cmd/ctrl + k</kbd>
       </sl-bar-item>
+
```
Comment (andrewmcodes) [inline bridgetown-website/src/_components/shared/navbar.serb:54]: Any reason for this newline?

**Comment by andrewmcodes** [comment]
@vvveebs this is looking awesome! I really only found one issue and then there are a few things I want to get your thoughts on:

1. The edge site banner appears to be the wrong color 
20221120211946Z_Safari_A Break from Summer Break to Bring You Bridgetown 0 21 1  Bridgetown  EDGE

2. The theme picker is excellent but it's a little weird that it doesn't match the style of the rest of the buttons. I think it's ok but one thing I think we should change is that the hover color is blue but it would be nice if it matched the hover color of the dropdown items aka the bridgetown green. Wdyt?

3. These buttons may look better without a border or with different shade of green for the border as opposed to white. Wdyt?
20221120212911Z_Safari_Getting Started  Bridgetown  EDGE

---

2 & 3 are opinions so not a requirement to merge. Overall I think this look great, thank you so much for the contribution 👏

**Comment by vvveebs** [comment]
@andrewmcodes thanks for the review! All good points. Responded to them in order:

1. [Edge site banner in dark mode updated](https://github.com/bridgetownrb/bridgetown/pull/620/commits/ebe9f2f2ed405f8dda8e05c913a8c78cc580d031).

image

2. It's funny - I didn't even realise this button was so different  - guess I'd been staring at it for a while and it became a blind spot! [Updated.](https://github.com/bridgetownrb/bridgetown/pull/620/commits/905aa9bfc06ae96030ac33150108df9fa34aa59c)

https://user-images.githubusercontent.com/1370570/204988341-c1baab23-acea-4763-ac77-c9aaa92b9c03.mov

3. After [chat with Jared](https://github.com/bridgetownrb/bridgetown/pull/620#discussion_r979299812) it seems like we want to go ahead with the border. But I agree straight up white is too much. [Updated to a light green.](https://github.com/bridgetownrb/bridgetown/pull/620/commits/ebe9dd158937e251d6e4223ff68c221b98ebbbac)

image

**Comment by jaredcwhite** [comment]
@vvveebs Looking gooood 🤩

**Comment by jaredcwhite** [comment]
@vvveebs I see some room for further tweaking, but I think it'll be better to reason about in followup issues/PRs. Going to go ahead and merge! Thanks again for this fabulous effort!

**Pull request merged**